receiver/rabbitmqreceiver/scraper.go

```
@@ -0,0 +1,205 @@
+		r.isOK(ok, "could not get consumers from body")
+		consumersVal, ok := r.parseFloat("consumers", consumersInter)
+		r.isOK(ok, "could not parse consumers as a float")
+		addToMetric(consumersMetric, labels, consumersVal, now)
+
+		numMessagesInter, ok := getValFromBody([]string{"messages"}, queue)
+		r.isOK(ok, "could not get messages from body")
+		numMessagesVal, ok := r.parseFloat("num_messages - messages", numMessagesInter)
+		labels.Upsert(metadata.L.State, "total")
+		r.isOK(ok, "could not parse messages as a float")
+		addToMetric(numMessagesMetric, labels, numMessagesVal, now)
+
+		numMessagesInter, ok = getValFromBody([]string{"messages_unacknowledged"}, queue)
+		r.isOK(ok, "could not get messages_unacknowledged from body")
+		numMessagesVal, ok = r.parseFloat("num_messages - messages_unacknowledged", numMessagesInter)
+		r.isOK(ok, "could not parse messages_unacknowledged as a float")
+		labels.Upsert(metadata.L.State, "unacknowledged")
+		addToMetric(numMessagesMetric, labels, numMessagesVal, now)
+
+		numMessagesInter, ok = getValFromBody([]string{"messages_ready"}, queue)
+		r.isOK(ok, "could not get messages_ready from body")
+		numMessagesVal, ok = r.parseFloat("num_messages - ready", numMessagesInter)
+		r.isOK(ok, "could not parse messages_ready as a float")
+		labels.Upsert(metadata.L.State, "ready")
+		addToMetric(numMessagesMetric, labels, numMessagesVal, now)
+	}
+
+	return metrics.ResourceMetrics(), nil
+}
+
+// parseFloat converts string to float64.
+func (r *rabbitmqScraper) parseFloat(key string, value interface{}) (float64, bool) {
+	switch f := value.(type) {
+	case float64:
+		return f, true
+	case int64:
+		return float64(f), true
+	case float32:
+		return float64(f), true
+	case int32:
+		return float64(f), true
+	case string:
+		fConv, err := strconv.ParseFloat(f, 64)
+		if err != nil {
+			r.logInvalid("float", key, f)
+			return 0, false
+		}
+		return fConv, true
+	}
+	return 0, false
+}
```

I would suggest making this just a normal function, and not a receiver on the scraper struct. It's a very simple and focused method, but the logging kind of complicates it. In my opinion, it'd be better to return the error, and log it from the caller.

---

done